cpp/src/parquet/metadata.h

```diff
@@ -171,6 +171,13 @@ class PARQUET_EXPORT ColumnChunkMetaData {
   int64_t total_uncompressed_size() const;
   std::unique_ptr<ColumnCryptoMetaData> crypto_metadata() const;
 
+  bool has_column_index() const;
+  int64_t column_index_offset() const;
```

Comment:
I think it might be a clearer contract to have something like:

```
struct IndexLocation {
   int64_t index_file_offset_bytes;
   int32_t offset_index_length
}

optional<IndexLocation> GetColumIndexLocation();
optional<IndexLocation> GetOffsetIndexLocation();
```

Thoughts?

Reply:
SGTM, I will change it shortly.